src/aggregated/circulating_supply/circulating_supply_provider.rs

```diff
@@ -0,0 +1,182 @@
+use std::ops::{Add, Div};
+use std::str::FromStr;
+use std::time::{Duration, SystemTime};
+
+use actix::Addr;
+use actix_diesel::Database;
+use actix_web::rt::time;
+use bigdecimal::{BigDecimal, ToPrimitive};
+use diesel::PgConnection;
+use tokio::time::Instant;
+use tracing::{error, info};
+
+use crate::aggregated::account_details;
+use crate::aggregated::account_details::get_account_balance;
+use crate::aggregated::circulating_supply::lockup;
+use crate::aggregated::circulating_supply::lockup::TRANSFERS_ENABLED;
+use crate::db_adapters::accounts;
+use crate::db_adapters::aggregated::circulating_supply::{
+    add_circulating_supply, get_precomputed_circulating_supply_for_timestamp,
+};
+use crate::db_adapters::blocks::get_latest_block_before_timestamp;
+use crate::models::aggregated::circulating_supply::CirculatingSupply;
+use crate::models::Block;
+
+const DAY: std::time::Duration = std::time::Duration::from_secs(86400);
+
+// Compute circulating supply on a daily basis, starting from 13 Oct 2020
+// (Transfers enabled moment on the Mainnet), and put it to the Indexer DB.
+// Circulating supply is calculated by the formula:
+// total_supply - sum(locked_tokens_on_each_lockup) - sum(locked_foundation_account)
+// The value is always computed for the last block in a day (UTC).
+pub(crate) async fn compute_circulating_supply(
```

Comment:
I fail to find a good name for this function since it does so many things at once. Some candidates

* `schedule_compute_circulating_supply` (it is not only scheduling, but keeps running)
* `compute_circulating_supply_daily` (it also back-fills, so it is not accurate)
* `aggregate_circulating_supply` :unamused: (it still fills like it is an action, but it is not going to be a single action)
* `handle_circulating_supply_in_background` :unamused:

Reply:
how about `run_circulating_supply_computation`?